File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/Order/Webstore.cs
/// <filename>
///     Webstore.cs
/// </filename>
using Order;

namespace Order
{
    internal class Webstore : Order.ISearchInterface, Order.IShoppingInterface
    {

        public Webstore()
        {
            this.searching = new Order.Searching();
            this.shoppingCart = new Order.ShoppingCart();
            this.inventory = new Order.Inventory();
            this.ordering = new Order.Ordering();
            this.shoppingCart.dependency_OrderManagement = this.ordering;
            this.searching.dependency_SearchInventory = this.inventory;
            this.ordering.dependency_ManageInventory = this.inventory;
        }

        public virtual int searchProduct(string productName)
        {
            return this.inventory.searchProduct(productName);
        }

        public virtual void addProduct(int productId)
        {
            this.shoppingCart.addProduct(productId);
        }

        public virtual void removeProduct(int productId)
        {
            this.shoppingCart.removeProduct(productId);
        }

        public virtual void orderProducts()
        {
            this.shoppingCart.orderProducts();
        }

        public Order.Searching searching;

        public Order.ShoppingCart shoppingCart;

        public Order.Inventory inventory;

        public Order.Ordering ordering;
    }
}

File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/Order/Webstore.cpp
#include "Webstore.h"
#include <memory>
#include "ShoppingCart.h"
#include "Searching.h"
#include "Ordering.h"
#include "ISearchInterface.h"
#include "IShoppingInterface.h"
#include "Inventory.h"
#include <string>

namespace Order
{
  Webstore::Webstore()
  {
    this->searching = std::make_shared<Searching>();
    this->shoppingCart = std::make_shared<ShoppingCart>();
    this->inventory = std::make_shared<Inventory>();
    this->ordering = std::make_shared<Ordering>();
    this->shoppingCart->dependency_OrderManagement = this->ordering;
    this->searching->dependency_SearchInventory = this->inventory;
    this->ordering->dependency_ManageInventory = this->inventory;
  }
  int Webstore::searchProduct(std::string productName)
  {
    return this->inventory->searchProduct(productName);
  }
  void Webstore::addProduct(int productId)
  {
    this->shoppingCart->addProduct(productId);
  }
  void Webstore::removeProduct(int productId)
  {
    this->shoppingCart->removeProduct(productId);
  }
  void Webstore::orderProducts()
  {
    this->shoppingCart->orderProducts();
  }
}
